### PR TITLE
ci: enable testing `Legacy Minpack (SciPy)` with the `simplifier` pass

### PR DESCRIPTION
**Pull request by kmr-srbh**

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I'll review it in a while.

**Comment by kmr-srbh** [comment]
I had verified the results by hand, they seem to match with that of GFortran.

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -495,47 +495,47 @@ jobs:
...
+            make
+            examples/example_hybrd
+            examples/example_hybrd1
+            examples/example_lmder1
+            examples/example_lmdif1
+            examples/example_primes
+            ctest
 
       # - name: Test Modern Minpack (Fortran-Lang)
       #   shell: bash -e -x -l {0}
```
Comment (Pranavchiku) [inline .github/workflows/CI.yml:541]: I believe this shall work as well.

Comment (Pranavchiku) [reply .github/workflows/CI.yml:541]: No, this doesn't work.

```console
% lfortran ./examples/example_hybrd.f90 --legacy-array-sections
[..]
code generation error: asr_to_llvm: module failed verification. Error:
Call parameter type does not match function signature!
double 0x3CB0000000000000
 double*  %3 = call double @_lcompilers_max0_f641(double* %epsfcn, double 0x3CB0000000000000)

```

**Review by Pranavchiku — APPROVED** [review]
This works, I tested it locally.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (kmr-srbh) [reply .github/workflows/CI.yml:541]: Okay, I will give it a try and open another PR. :+1:

Comment (kmr-srbh) [reply .github/workflows/CI.yml:541]: I think the above failure is caused due to changes made in #4753.

**Pull request merged**